src/battle_util.c

```diff
@@ -7664,3 +7683,17 @@ bool32 SetIllusionMon(struct Pokemon *mon, u32 battlerId)
 
     return FALSE;
 }
+
+static bool8 ShouldGetStatBadgeBoost(u16 badgeFlag, u8 battlerId)
```

Comment:
You can add `if (B_BADGE_BOOST != GEN_3) return FALSE;` to this function, so that you won't have to check each time you call this function. This would make the code simpler in every instance you call `ShouldGetStatBadgeBoost` because you won't have to preface it with an if because it would already be handled.